api/content.go

```diff
@@ -11,55 +12,103 @@ import (
+	staticdatasetsStr      = "staticdatasets"
+	timeseriesStr          = "timeseries"
+	publicationsStr        = "publications"
+	datasetsStr            = "datasets"
+)
+
+var querySets map[string]int = map[string]int{
+	// search keys are done as lower case to make searches work regardless of case
+	spotlightStr:           querySpotlight,
+	articlesStr:            queryAarticles,
+	bulletinsStr:           queryBulletins,
+	methodologiesStr:       queryMethodologies,
+	methodologyarticlesStr: queryMethodologyArticles,
+	staticdatasetsStr:      queryStaticDatasets,
+	timeseriesStr:          queryTimeseries,
+
+	publicationsStr: queryAarticles | queryBulletins | queryMethodologies | queryMethodologyArticles,
+
+	datasetsStr: queryStaticDatasets | queryTimeseries,
+}
+
+// getContentTypeParameter obtains a filter that defines a set of possible types
+func getContentTypeParameter(queryVars url.Values) int {
+	valArray, found := queryVars["type"]
+	if !found {
+		// no type specified, so return flags for all types
+		return querySpotlight | queryAarticles | queryBulletins | queryMethodologies | queryMethodologyArticles | queryStaticDatasets | queryTimeseries
 	}
 
-	nofItems := len(*itemLink)
-	if nofItems == 0 {
-		return
+	// make query type lower case for following comparison to cope with wrong case of letter(s)
+	lowerVal := strings.ToLower(valArray[0])
+
+	// also remove leading and trailing whitespace as it casuses the check to fail
+	trimmedVal := strings.TrimSpace(lowerVal)
+
+	set, ok := querySets[trimmedVal]
+	if ok {
+		return set
 	}
 
-	title := make(map[string]string)
-
-	// Create list of sorted href's from itemLink list
-	hrefs := make([]string, nofItems)
-	for i, field := range *itemLink {
-		hrefs[i] = field.HRef
-		title[field.HRef] = field.Title
-	}
-	sort.Strings(hrefs)
-
-	// Iterate through alphabeticaly sorted 'hrefs' and use each one to select corresponding title
-	for _, href := range hrefs {
-		// build up data items into structure
-		var cItem models.ContentItem = models.ContentItem{
-			Title: title[href],
-			Type:  typeName,
-			Links: &models.ContentLinks{
-				Self: &models.LinkObject{
-					HRef: href,
-				},
-				Topic: &models.LinkObject{
-					ID:   id,
-					HRef: "/topic/" + id,
-				},
-			},
-		}
-
-		if privateResponse {
-			cItem.State = state
-		}
-
-		if contentList.Items == nil {
-			contentList.Items = &[]models.ContentItem{cItem}
-		} else {
-			*contentList.Items = append(*contentList.Items, cItem)
-		}
-	}
-
-	contentList.TotalCount = contentList.TotalCount + nofItems
+	return 0
+}
+
+func getRequiredItems(queryType int, content *models.Content, id string) *models.ContentResponseAPI {
+	var result models.ContentResponseAPI
+
+	// Add spotlight first
+	if (queryType & querySpotlight) != 0 {
+		result.AppendLinkInfo(spotlightStr, content.Spotlight, id, content.State)
+	}
+
+	// then Publications (alphabetically ordered)
+	if (queryType & queryAarticles) != 0 {
```

Comment:
queryAarticles there's a double A in this var name thats not needed

Reply:
done